src/qt/intro.cpp

```diff
@@ -349,3 +332,39 @@ QString Intro::getPathToCheck()
     mutex.unlock();
     return retval;
 }
+
+void Intro::setSizeWarningLabel(bool isPruning, uint64_t blockchain_size, uint64_t chain_state_size)
+{
+    requiredSpace = blockchain_size;
+    QString storageRequiresMsg = QObject::tr("At least %1 GB of data will be stored in this directory, and it will grow over time.");
+    if (isPruning) {
+        uint64_t pruneTarget = std::max<int64_t>(0, gArgs.GetArg("-prune", 0));
+        uint64_t prunedGBs = std::ceil(pruneTarget * 1024 * 1024.0 / GB_BYTES);
+        if (prunedGBs <= requiredSpace) {
+            requiredSpace = prunedGBs;
+            storageRequiresMsg = QObject::tr("Approximately %1 GB of data will be stored in this directory.");
+        }
+        Intro::ui->lblExplanation3->setVisible(true);
+    } else {
+        Intro::ui->lblExplanation3->setVisible(false);
+    }
+    requiredSpace += chain_state_size;
+    Intro::ui->sizeWarningLabel->setText(
+        QObject::tr("%1 will download and store a copy of the Bitcoin block chain.").arg(PACKAGE_NAME) + " " +
+        storageRequiresMsg.arg(requiredSpace) + " " +
+        QObject::tr("The wallet will also be stored in this directory.")
+    );
+    // Reset GUI width and update storage warning if neccesary
+    this->adjustSize();
+}
+void Intro::on_prune_stateChanged(int arg1)
```

Comment:
IMO we should move away from these dynamic connections. My suggestion is to move to the constructor:
```cpp
connect(ui->prune, &QCheckBox::stateChanged, [this](int state) {
    setSizeWarningLabel(/* isPruning = */ state == 2, m_blockchain_size, m_chain_state_size);
});
```

Reply:
I agree with you but this is a bigger change and should be a done in a separate PR because this one is intended to be a fix, not a replacement

Reply:
Huh? Bigger? Please unresolved this.